packages/flutter_tools/lib/src/base/process_manager.dart

```diff
@@ -479,3 +492,296 @@ class _RecordingProcess implements Process {
+      String executable,
+      List<String> arguments,
+      {String workingDirectory,
+       Map<String, String> environment,
+       Encoding stdoutEncoding: SYSTEM_ENCODING,
+       Encoding stderrEncoding: SYSTEM_ENCODING}) async {
+    Map<String, dynamic> entry = _popEntry(executable, arguments,
+        stdoutEncoding: stdoutEncoding, stderrEncoding: stderrEncoding);
+    return await _ReplayProcessResult.create(_dir, entry);
+  }
+
+  @override
+  ProcessResult runSync(
+      String executable,
+      List<String> arguments,
+      {String workingDirectory,
+       Map<String, String> environment,
+       Encoding stdoutEncoding: SYSTEM_ENCODING,
+       Encoding stderrEncoding: SYSTEM_ENCODING}) {
+    Map<String, dynamic> entry = _popEntry(executable, arguments,
+        stdoutEncoding: stdoutEncoding, stderrEncoding: stderrEncoding);
+    return _ReplayProcessResult.createSync(_dir, entry);
+  }
+
+  /// Finds and returns the next entry in the process manifest that matches
+  /// the specified process arguments. Once found, it marks the manifest entry
+  /// as having been invoked and thus not eligible for invocation again.
+  Map<String, dynamic> _popEntry(String executable, List<String> arguments, {
+    ProcessStartMode mode,
+    Encoding stdoutEncoding,
+    Encoding stderrEncoding,
+  }) {
+    Map<String, dynamic> entry = _manifest.firstWhere(
+      (Map<String, dynamic> entry) {
+        // Ignore workingDirectory & environment, as they could
+        // yield false negatives.
+        return entry['executable'] == executable
+            && kListsEqual(entry['arguments'], arguments)
+            && entry['mode'] == mode?.toString()
+            && entry['stdoutEncoding'] == stdoutEncoding?.name
+            && entry['stderrEncoding'] == stderrEncoding?.name
+            && !(entry['invoked'] ?? false);
+      },
+      orElse: () => null,
+    );
+
+    if (entry == null)
+      throw new StateError('No matching invocation found for $executable');
+
+    entry['invoked'] = true;
+    return entry;
+  }
+
+  @override
+  bool killPid(int pid, [ProcessSignal signal = ProcessSignal.SIGTERM]) {
+    throw new UnsupportedError(
+        "$runtimeType.killPid() has not been implemented because at the time "
+        "of its writing, it wasn't needed. If you're hitting this error, you "
+        "should implement it.");
+  }
+}
+
+/// A [ProcessResult] implementation that derives its data from a recording
+/// fragment.
+class _ReplayProcessResult implements ProcessResult {
+  @override
+  final int pid;
+
+  @override
+  final int exitCode;
+
+  @override
+  final dynamic stdout;
+
+  @override
+  final dynamic stderr;
+
+  _ReplayProcessResult._({this.pid, this.exitCode, this.stdout, this.stderr});
+
+  static Future<_ReplayProcessResult> create(
+    Directory dir,
+    Map<String, dynamic> entry,
+  ) async {
+    String basePath = path.join(dir.path, entry['basename']);
+    return new _ReplayProcessResult._(
```

Comment:
The failure path here where _getData throws an exception (because stdout or stderr can't be found) might bubble up to the callers. Consider catching here and throwing whatever error Process throws when it can't spawn a process.